roles/DCOS.agent_windows/tasks/main.yml

```diff
@@ -47,3 +47,37 @@
     dependencies: [ 'docker' ]
     start_mode: auto
   notify: restart the service
+
+- name: Set an environment variable for all users
+  win_environment:
+    state: present
+    name: DCOS_VERSION
+    value: "{{ dcos['version'] }}"
+    level: machine
+
+
+- name: Check that dcos-diagnostics.exe exists
+  win_stat:
+    path: C:\dcos\dcos-diagnostics.exe
+  register: stat_result
+
+- name: Windows | Download dcos-diagnostics
+  win_get_url:
+    url: "{{ url_to_diagnostics }}"
+    dest: C:\dcos\dcos-diagnostics.zip
+    force: no
+  when: stat_result.stat.exists == False 
```

Comment:
`when: not stat_result.stat.exists`

Reply:
fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580